lib/tzinfo/time_or_datetime.rb

```diff
@@ -255,6 +255,29 @@ def -(seconds)
       self + (-seconds)
     end
 
+    # Sets offset from UTC for the TimeOrDateTime. Returns a new TimeOrDateTime,
+    # preserving original class. Does NOT changes values of hour-min-second,
+    # so, (18:00 UTC).offset(3600) will became (18:00 +01:00).
+    # Considers original value's UTC offset wisely.
+    def offset(seconds)
+      return self if seconds == 0
+
+      if @orig.is_a?(DateTime)
+        off = OffsetRationals.rational_for_offset(seconds)
+        TimeOrDateTime.new(@orig.new_offset(off), false)
+      elsif @orig.is_a?(Time)
+        offset_str = '%+03i:%02i' % [seconds / 3600, seconds / 60 % 60]
```

Comment:
It looks like the `Time` constructor can take either a `String` or an `Integer` number of seconds as the offset. Would it be better to avoid the conversion to a `String`?